trussworks/react-uswds
# feat: Update Table to 2.10.0 implementation

# Summary

This PR updates the [USWDS Table component](https://designsystem.digital.gov/components/table) with `scrollable`, `striped`, `compact`, `stacked`, and `stacked-header` variations as added in the [USWDS 2.10.0 release](https://designsystem.digital.gov/about/releases/#version-2100).

## Related Issues or PRs

closes #991 

* Is there a better way than putting `stackedStyle = 'none'` in the function signature to assign a default value?

This PR does not add the [Sortable Table rows](https://designsystem.digital.gov/components/table/#sortable-table-rows) variant as that is future work to cover updates made by the USWDS 2.11.0 release.

Reviewers may notice some styling differences between our Storybook versus the USWDS examples, particularly with regards to fonts/text sizing. As [Hana has noted](https://github.com/trussworks/react-uswds/pull/1110#issuecomment-820733923) further down in this PR, we suspect this is due to the difference in how USWDS applies their CSS to how we apply it, and is less relevant to this work as we are simply rendering those applied styles on children, not the Table component itself. 

## How To Test

Check out this branch and run `yarn test` to execute tests and `yarn storybook` to see the new component in action. 

Alternatively, you can scroll to the bottom of this page and click "Show environments" on the right above the comment input box to access this component in Storybook.

### Screenshots

#### Scrollable Table
![scrollable-table](https://user-images.githubusercontent.com/16230705/114945018-7038f600-9dfd-11eb-8fd2-776ddf0b03f9.gif)




#### Striped Table

<img width="955" alt="Screen Shot 2021-04-15 at 3 10 05 PM" src="https://user-images.githubusercontent.com/16230705/114944562-ac1f8b80-9dfc-11eb-8211-61d7e9cac1a4.png">



#### Compact Table

<img width="954" alt="Screen Shot 2021-04-15 at 3 10 54 PM" src="https://user-images.githubusercontent.com/16230705/114944613-c9545a00-9dfc-11eb-9a11-af9258bd70e7.png">



#### Logged warning if the user attempts to apply both `compact` and `stackedStyle`

From the USWDS notes re: compact styling: "Pairs well with scrollable and striped variants, but is not suitable for use with stacked variants."

![Screen Shot 2021-04-13 at 2 52 12 PM](https://user-images.githubusercontent.com/16230705/114626031-fde2dd00-9c67-11eb-9efa-1d2f5d34d100.png)



#### Stacked Table With Custom Controls `bordered` and `stackedStyle`

![stackedStyles](https://user-images.githubusercontent.com/16230705/114945354-04a35880-9dfe-11eb-97fe-c70a6faadcdd.gif)
